E0094 needs to be updated to new format

From: src/test/compile-fail/E0094.rs

Error E0094 needs a span_label, updating it from:

```
error[E0094]: intrinsic has wrong number of type parameters: found 2, expected 1
  --> src/test/compile-fail/E0094.rs:13:5
   |
13 |     fn size_of<T, U>() -> usize; //~ ERROR E0094
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

To:

```
error[E0094]: intrinsic has wrong number of type parameters: found 2, expected 1
  --> src/test/compile-fail/E0094.rs:13:5
   |
13 |     fn size_of<T, U>() -> usize; //~ ERROR E0094
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 1 type parameter
```

Bonus: That span is pretty long, though I'm not sure if it can be fixed.  Worth seeing if we can shrink the span to:

```
error[E0094]: intrinsic has wrong number of type parameters: found 2, expected 1
  --> src/test/compile-fail/E0094.rs:13:5
   |
13 |     fn size_of<T, U>() -> usize; //~ ERROR E0094
   |                ^^^^ expected 1 type parameter
```

--- a/src/librustc_typeck/check/intrinsic.rs
+++ b/src/librustc_typeck/check/intrinsic.rs
@@ -51,10 +51,12 @@ fn equate_intrinsic_type<'a, 'tcx>(ccx: &CrateCtxt<'a, 'tcx>,
     }));
     let i_n_tps = i_ty.generics.types.len(subst::FnSpace);
     if i_n_tps != n_tps {
-        span_err!(tcx.sess, it.span, E0094,
+        struct_span_err!(tcx.sess, it.span, E0094,
             "intrinsic has wrong number of type \
              parameters: found {}, expected {}",
-             i_n_tps, n_tps);
+             i_n_tps, n_tps)
+             .span_label(it.span, &format!("expected {} type parameter", n_tps))
+             .emit();
     } else {
         require_same_types(ccx,
                            TypeOrigin::IntrinsicType(it.span),

--- a/src/test/compile-fail/E0094.rs
+++ b/src/test/compile-fail/E0094.rs
@@ -11,6 +11,7 @@
 #![feature(intrinsics)]
 extern "rust-intrinsic" {
     fn size_of<T, U>() -> usize; //~ ERROR E0094
+                                 //~| NOTE expected 1 type parameter
 }
 
 fn main() {

